src/server/protocol.ts

```diff
@@ -2174,6 +2187,8 @@ namespace ts.server.protocol {
          * Names of one or more entries for which to obtain details.
          */
         entryNames: (string | CompletionEntryIdentifier)[];
+        /** if true - return response with documentation as display parts instead of string */
+        richResponse?: boolean;

```

Comment:
I realize this is kind of late, but completion has so many switches that "rich" isn't very clarifying.  At the very least, it should be "rich documentation".  Also, probably a user preference, as above.